ui/src/app/components/sharded-dataflow-migration-details-form/sharded-dataflow-migration-details-form.component.html

```diff
@@ -0,0 +1,211 @@
+                <mat-form-field appearance="outline">
+                    <mat-label>Data Shard Id</mat-label>
+                    <input matInput placeholder="Data ShardId" type="text" formControlName="dataShardId"
+                        required="true" />
+                </mat-form-field>
+            </div>
+            <div *ngIf="(selectedSourceProfileOption === 'new')">
+                <mat-form-field appearance="outline">
+                    <mat-label>Host</mat-label>
+                    <input matInput placeholder="Host" type="text" formControlName="host" required="true" />
+                </mat-form-field>
+                <mat-form-field appearance="outline">
+                    <mat-label>User</mat-label>
+                    <input matInput placeholder="User" type="text" formControlName="user" required="true" />
+                </mat-form-field>
+                <mat-form-field appearance="outline">
+                    <mat-label>Port</mat-label>
+                    <input matInput placeholder="Port" type="text" formControlName="port" required="true" />
+                </mat-form-field>
+                <mat-form-field appearance="outline">
+                    <mat-label>Password</mat-label>
+                    <input matInput placeholder="Password" type="password" formControlName="password" required="true" />
+                </mat-form-field>
+                <mat-form-field appearance="outline">
+                    <mat-label>Connection profile name</mat-label>
+                    <input matInput placeholder="Connection profile name" type="text" formControlName="newSourceProfile"
+                        required="true" />
+                </mat-form-field>
+                <mat-form-field appearance="outline">
+                    <mat-label>Data Shard Id</mat-label>
+                    <input matInput placeholder="Data ShardId" type="text" formControlName="dataShardId"
+                        required="true" />
+                </mat-form-field>
+            </div>
+            <div *ngIf="(selectedSourceProfileOption === 'new')">
+                <div><b>Copy the public IPs below, and use them to configure the network firewall to accept
+                        connections
+                        from them.</b><a
+                        href="https://cloud.google.com/datastream/docs/network-connectivity-options#ipallowlists">Learn
+                        More</a></div>
+                <br>
+                <li *ngFor="let ip of ipList" class="connection-form-container">
+                    <span class="left-text">{{ip}}</span>
+                    <span>
+                        <mat-icon class="icon copy" [cdkCopyToClipboard]="ip" matTooltip="Copy">file_copy</mat-icon>
+                    </span>
+                </li>
+                <div *ngIf="!testSuccess && errorSrcMsg != ''" class="failure">
+                    <span class="left-text">Test connection failed</span>
```

Comment:
Didn't get anything after clicking on test connection and then clicked on create connection profile and got test failed , create profile shouldn't validate the connection therefore it should be disabled if the connection is not valid.

Reply:
There was an error in the validation. Fixed now.